### DISCUSSION TITLE
how to enable extension(vsix) custom logging output channel in theia app

### DISCUSSION THREAD

**Discussion by omkar07** [eclipse-theia/theia#9735] [Q&A]
hi team, I'm loading vscode extension vsix file in which we have created custom logger output channel to log all logs for the extension in [theia app](https://theia-ide.org/docs/composing_applications). When I load this extension in vscode i'm able to see the output channel but for theia I don't see this logger channel in output window. I didn't find any resource or documentation on it. could you please provide any suggestions?
<img width="1066" alt="Screen Shot 2021-07-13 at 8 06 11 PM" src="https://user-images.githubusercontent.com/19549743/125554647-18e1a9f4-e66d-4c12-8335-94da28121a1f.png">

thanks!

**Comment by vince-fugnitto** [comment]
@omkar07 thank you for the discussion, do you mind sharing which extension you tried it for, was it `EditorConfig`?

I confirmed using a custom vscode plugin (which uses the `vscode.window.createOutputChannel`) that output logging from extensions works:

![image](https://user-images.githubusercontent.com/40359487/125620650-e8cf1ed6-9319-40e1-a0a2-ac831112974a.png)

**Reply by omkar07** [reply]
yes, it's working but it seems it doesn't show logs when the command executes and the output window is displayed for first time. I had to close the output window and then reopen it from the view tab to see the written logs. I tried with sample helloworld extension with the custom channel. Was the behaviour the same for you?
[helloworld-sample-0.0.1.vsix.zip](https://github.com/eclipse-theia/theia/files/7348490/helloworld-sample-0.0.1.vsix.zip)

**Reply by vince-fugnitto** [reply]
@omkar07 unfortunately, the plugin worked as expected for me:
- if the output-view is not opened, executing the command will append lines to the channel (opening the view will display the output properly)
- if the output-view is opened (before ever executing the command), executing the command will append lines to the channel

If the issue persists for you can you describe your 'steps to reproduce' thoroughly, ideally with a screenshare (video) if possible?
